test/users-query.test.ts

```diff
@@ -0,0 +1,178 @@
+      },
+    ],
+
+    negativeAmountError: [
+      {
+        code: 400,
+        message: "Amount of users must be greater or equal to 0",
+      },
+    ],
+  };
+
+  const createUsersQuery = (amount?: number, usersToSkip?: number) => {
+    const queryBody = {
+      query: `query users($amount: Int, $usersToSkip: Int) {
+		users(amount: $amount, usersToSkip: $usersToSkip) {
+		  currentPage
+		  numberOfPages
+		  numberOfUsers
+		  users {
+			dateOfBirth
+			email
+			id
+			name
+			profession
+		  }
+		}
+	  }`,
+      variables: {
+        amount: amount,
+        usersToSkip: usersToSkip,
+      },
+    };
+    return queryBody;
+  };
+
+  it("should return the correct array of users", async () => {
+    await fillDatabase(42);
+    const dbUser = await UserRepository.findOne({
+      where: { email: "test@test.com" },
+    });
+
+    const testUsers: UserPayload[] = await UserRepository.find({
+      order: {
+        name: "ASC",
+      },
+      skip: 3,
+      take: 10,
+    });
+
+    const testPagedUsers: PagedUser[] = testUsers.map((users) => ({
+      ...users,
+      dateOfBirth: users.dateOfBirth.toISOString(),
+    }));
+    testPagedUsers.forEach((user) => {
+      delete user["password"];
+    });
+
+    const queryBody = createUsersQuery(10, 3);
+    const queryResponse = await getQuery(queryBody, getToken(dbUser, true));
+    expect(queryResponse.data.data.users.users).to.deep.equal(testPagedUsers);
+  });
+
+  it("should return the correct amount of total users", async () => {
+    await fillDatabase(42);
+    const dbUser = await UserRepository.findOne({
+      where: { email: "test@test.com" },
+    });
+    const queryBody = createUsersQuery(10, 3);
+    const queryResponse = await getQuery(queryBody, getToken(dbUser, true));
+    expect(queryResponse.data.data.users.numberOfUsers).to.equal(43);
+  });
+
+  it("should return the correct amount of pages", async () => {
+    await fillDatabase(42);
+    const dbUser = await UserRepository.findOne({
+      where: { email: "test@test.com" },
+    });
+    const queryBody = createUsersQuery(10, 3);
+    const queryResponse = await getQuery(queryBody, getToken(dbUser, true));
+    expect(queryResponse.data.data.users.numberOfPages).to.equal(5);
+  });
+
+  it("should return the correct number for current page", async () => {
+    await fillDatabase(42);
+    const dbUser = await UserRepository.findOne({
+      where: { email: "test@test.com" },
+    });
+    const queryBody = createUsersQuery(10, 3);
+    const queryResponse = await getQuery(queryBody, getToken(dbUser, true));
+    expect(queryResponse.data.data.users.currentPage).to.equal(2);
+  });
+
+  it("should throw error of sum of amount and skipped users greater than total users on database", async () => {
+    await fillDatabase(42);
+    const dbUser = await UserRepository.findOne({
+      where: { email: "test@test.com" },
+    });
+    const queryBody = createUsersQuery(50, 3);
```

Comment:
aqui você testa o caso do `amount` ser maior que o número de usuários no banco, e o caso do `skip`?

Reply:
Eu testo a soma, os dois casos resultariam em uma página com menos usuários que o requisitado.

Reply:
sim, mas que tal testar explicitamente cada um, apesar do resultado ser o mesmo, agora você está garantindo que cado o `amount` seja maior vai dar erro, mas esse teste não me indica que se o `skip` for maior vai dar tb